server/pulp/server/managers/consumer/applicability.py

```diff
@@ -152,6 +152,7 @@ def queue_regenerate_applicability_for_repos(repo_criteria):
                     if len(profiles_to_process) < batch_size:
                         profiles_to_process.append((repo_id, all_profiles_hash, profiles))
                     else:
+                        profiles_to_process.append((repo_id, all_profiles_hash, profiles))
                         batch_regenerate_applicability_task.apply_async(
                             (profiles_to_process,), **{'group_id': task_group_id})
                         profiles_to_process = []
```

Comment:
It seems like the same code is in both condition statements, should it just be outside?
```
profiles_to_process.append((repo_id, all_profiles_hash, profiles))
if len(profiles_to_process) >= batch_size:
    batch_regenerate_applicability_task.apply_async(
        (profiles_to_process,), **{'group_id': task_group_id})
￼   profiles_to_process = []
```
@hao-yu , what do you think?
And thanks for the patch!

Reply:
Yeah, that looks good to me. I will make the change and re-submit the patch. Thanks for the suggestion.